runtime/doc/quickfix.txt

```diff
@@ -1921,6 +1922,63 @@ error messages into a format that quickfix mode will understand.  See the
 start of the file about how to use it.  (This script is deprecated, see
 |compiler-perl|.)
 
-
+=============================================================================
+10. Customizing the quickfix window		*quickfix-window-function*
+
+The default format for the lines displayed in the quickfix window and location
+list window is:
+
+    <filename>|<lnum> col <n>|<text>
+
+This can be customized by setting the 'quickfixtextfunc' option to a Vim
+function.  This function will be called with a dict argument for every entry
+in the quickfix list. The dict argument will have the following fields:
+
+    quickfix	set to 1 when called for a quickfix list
+    id		quickfix or location list identifier
+    idx		index of the entry in the quickfix or location list
+
+The function should return a single line of text to display in the quickfix
+window for the entry identified by idx.
+
+If a quickfix or location list specific customization is needed, then the
+'quickfixtextfunc' attribute of the list can be set using the |setqflist()| or
+|setloclist()| function. This overrides the 'quickfixtextfunc' option setting.
+
+The example below tries to mimic the default behavior for the quickfix
+and location list windows.
+Example: >
```

Comment:
i suspect this example is too complex. Is it better to use a simpler, motivating example? Like why this feature was added in a nutshell?

Reply:
To keep most of the contents of the quickfix/location list buffer but only modify
selected components like path name or other parts of the line, you still need to
handle all the cases. I will add comments to this example.